### DISCUSSION TITLE
Error while importing PYVISTA

### DISCUSSION THREAD

**Discussion by sumantkrsoni** [pyvista/pyvista#5438] [Q&A] [answered]
Hello Everyone!!!

I am getting an error while importing "pyvista".

Here is the list of errors. Please help me to fix it.

=============================================

>>> import pyvista
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\suman\miniconda3\envs\pyt\lib\site-packages\pyvista\__init__.py", line 9, in <module>
    from pyvista.core import *
  File "C:\Users\suman\miniconda3\envs\pyt\lib\site-packages\pyvista\core\__init__.py", line 4, in <module>
    from . import _vtk_core
  File "C:\Users\suman\miniconda3\envs\pyt\lib\site-packages\pyvista\core\_vtk_core.py", line 309, in <module>
    from vtkmodules.vtkFiltersParallelDIY2 import vtkRedistributeDataSetFilter
ImportError: Initialization failed for vtkFiltersParallelDIY2, not compatible with vtkmodules.vtkCommonMath

============================================================

**Comment by banesullivan** [comment]
Would you please share which version of VTK you have in your environment and how you installed it (pip or conda)?

**Reply by sumantkrsoni** [reply]
It is showing "VTK 9.2.6" and it is installed through the pip.

please suggest to fix it.

Moreover, may I ask a doubt related to the streamline plotting in PYVISTA?

**Reply by banesullivan** [reply]
This was supposed to have been fixed in #4054... but it looks like #4018 may have incorrectly reverted it https://github.com/pyvista/pyvista/pull/4018/files#diff-d82c7cce72319a375b24846ad84d5e38b7cf205063148523e289bc0cf8c1c406R315-R319

I will post a patch now

**Reply by banesullivan** [reply, marked as answer]
Fix in https://github.com/pyvista/pyvista/pull/5467

**Comment by haennah** [comment]
Hello,

I am encountering a similar issue when trying to import PyVista:

ImportError: cannot import name 'MatrixLike' from 'pyvista.core._typing_core'

Error Traceback:
=============================================
File ~\AppData\Roaming\Python\Python312\site-packages\pyvista\__init__.py:14
     12 from pyvista._version import __version__
     13 from pyvista._version import version_info
---> 14 from pyvista.core import *
     15 from pyvista.core import _validation
     16 from pyvista.core._vtk_core import vtk_version_info

File ~\AppData\Roaming\Python\Python312\site-packages\pyvista\core\__init__.py:8
      6 from . import _vtk_core
      7 from ._typing_core import *
----> 8 from .cell import Cell
      9 from .cell import CellArray
     10 from .celltype import CellType

File ~\AppData\Roaming\Python\Python312\site-packages\pyvista\core\cell.py:16
     14 from . import _vtk_core as _vtk
...
---> 22 from ._typing_core import MatrixLike
     23 from ._typing_core import Number
     24 from ._typing_core import NumpyArray

ImportError: cannot import name 'MatrixLike' from 'pyvista.core._typing_core' 
=============================================

Environment Details:
- pyvista==0.43.8
- vtk==9.3.0
- Python 3.12.3

Any suggestions on how to resolve this?

Thank you for your help!

**Reply by user27182** [reply]
> Hello,
> 
> I am encountering a similar issue when trying to import PyVista:
> 
> ImportError: cannot import name 'MatrixLike' from 'pyvista.core._typing_core'
>
> Environment Details:
> 
> * pyvista==0.43.8
> * vtk==9.3.0
> * Python 3.12.3
> 

I just tried running:
``` shell
pip install pyvista==0.43.8 vtk==9.3.0
python -c "import pyvista as pv; print(pv.Report())"
```
And it ran without issue for me locally. Can you try re-installing in a new environment? Or provide more details about your current environment?

**Reply by haennah** [reply]
Hello, I tried re-installing in a new environment but the issue persists. My environment is attached here. Thank you for any help.
[requirements.txt](https://github.com/user-attachments/files/18645142/requirements.txt)

**Reply by haennah** [reply]
Update: The code runs without any issue in Jupyter lab but I still encounter issues when running it in VSCode.
